File: java/src/main/java/org/dilithia/sdk/model/TxHash.java
package org.dilithia.sdk.model;

/**
 * Strongly-typed wrapper for a transaction hash.
 *
 * <p>Transaction hashes are non-blank hex strings that uniquely identify
 * a transaction on the Dilithia network.</p>
 *
 * <p>Gson serialization is handled by a custom {@code TypeAdapter} registered
 * in {@link org.dilithia.sdk.internal.Json}.</p>
 *
 * @param value the transaction hash string
 */
public record TxHash(String value) {

    /**
     * Validates that the transaction hash is non-null and non-blank.
     *
     * @throws IllegalArgumentException if {@code value} is null or blank
     */
    public TxHash {
        if (value == null || value.isBlank()) throw new IllegalArgumentException("txHash must not be blank");
    }

    /**
     * Static factory for creating a {@code TxHash} from a plain string.
     *
     * @param value the transaction hash string
     * @return a new {@code TxHash}
     */
    public static TxHash of(String value) { return new TxHash(value); }

    @Override public String toString() { return value; }
}

File: dotnet/Dilithia.Sdk/Models/TxHash.cs
namespace Dilithia.Sdk.Models;

/// <summary>
/// Strongly-typed wrapper for a transaction hash.
/// </summary>
public readonly record struct TxHash(string Value)
{
    /// <summary>Create a <see cref="TxHash"/> from a raw string.</summary>
    public static TxHash Of(string value) => new(value);

    /// <inheritdoc />
    public override string ToString() => Value;

    /// <summary>Implicit conversion to string.</summary>
    public static implicit operator string(TxHash h) => h.Value;
}
